packages/hint-compat-api/src/helpers/compat-api.ts

```diff
@@ -0,0 +1,156 @@
+        if (!browserFeatureSupported) {
+            return browserFeatureSupported;
+        }
+
+        // Take the smaller version_removed and bigger version_added
+        const worstBrowserFeatureSupported: SimpleSupportStatement = {
+            version_added: null,
+            version_removed: null
+        };
+
+        if (Array.isArray(browserFeatureSupported) && browserFeatureSupported.length > 0) {
+            browserFeatureSupported.forEach((info) => {
+                if (!worstBrowserFeatureSupported.version_added && info.version_added === true) {
+                    worstBrowserFeatureSupported.version_added = true;
+                }
+
+                if (!worstBrowserFeatureSupported.version_added || worstBrowserFeatureSupported.version_added && info.version_added && info.version_added > worstBrowserFeatureSupported.version_added) {
+                    worstBrowserFeatureSupported.version_added = info.version_added;
+                }
+
+                if (!worstBrowserFeatureSupported.version_removed && info.version_removed === true) {
+                    worstBrowserFeatureSupported.version_removed = true;
+                }
+
+                if (!worstBrowserFeatureSupported.version_removed || worstBrowserFeatureSupported.version_removed && info.version_removed && info.version_removed < worstBrowserFeatureSupported.version_removed) {
+                    worstBrowserFeatureSupported.version_removed = info.version_removed;
+                }
+            });
+
+            return worstBrowserFeatureSupported;
+        }
+
+        return browserFeatureSupported as SimpleSupportStatement;
+    }
+    /* eslint-enable camelcase */
+
+    private isFeatureRequiredToTest(typedFeatureValue: CompatStatement & MDNTreeFilteredByBrowsers, isCheckingNotBroadlySupported = false): boolean {
+        // TODO: Here we are checking only parent but this object has children
+        let isRequiredToTest = false;
+
+        forEach(this.browsers, (browserVersionsList, browser) => {
+            if (isRequiredToTest || !typedFeatureValue.__compat || !typedFeatureValue.__compat.support) {
+                return;
+            }
+
+            const browserFeatureSupported = this.getWorstCaseSupportStatementFromInfo((typedFeatureValue.__compat.support as any)[browser]);
+
+            // If we dont have information about the compatibility, ignore.
+            if (!browserFeatureSupported) {
+                return;
+            }
+
+            const { version_added: addedVersion, version_removed: removedVersion } = browserFeatureSupported;
+
+            if (isCheckingNotBroadlySupported) {
+                if (!addedVersion || isNaN(parseFloat(addedVersion.toString()))) {
+                    return;
+                }
+
+                if (addedVersion || browserVersionsList[0] <= browserVersions.normalize(addedVersion)) {
+                    isRequiredToTest = true;
+                }
```

Comment:
Maybe add a `return` here so we can remove the else in line 143 and have the next `if`s with less nesting?